runtime/vm/linearswalk.c

```diff
@@ -427,21 +433,30 @@ lswDescribeSlot(J9VMThread * vmThread, J9StackWalkState * walkState, J9SWSlot *
 		case LSW_TYPE_INDIRECT_O_SLOT:
 		case LSW_TYPE_O_SLOT:
 			if (slot->data) {
-			
+				char *newPtr = NULL;
 				if (slot->type == LSW_TYPE_INDIRECT_O_SLOT) {
+					int len = 0;
 					object = (J9Object *) *((UDATA *) (slot->data & ~1));
-					dataPtr += sprintf(dataPtr, "%p -> ", object);
+					len = snprintf(dataPtr, dataSize, "%p -> ", object);
+					if ((0 < len) && ((size_t)len <= dataSize)) {
+						dataPtr += len;
+						dataSize -= len;
+					} else {
+						dataSize = 0;
+					}
 				} else {
 					object = (J9Object *) slot->data;
 				}
 				
 				object = (J9Object *) object;
 				class = J9OBJECT_CLAZZ(vmThread, object);
 				
-				dataPtr = lswDescribeObject(vmThread, dataPtr, &class, object);
+				newPtr = lswDescribeObject(vmThread, dataPtr, dataSize, &class, object);
+				dataSize -= newPtr - dataPtr;
+				dataPtr = newPtr;
 
 				className = J9ROMCLASS_CLASSNAME(class->romClass);
-				copySize = (J9UTF8_LENGTH(className) <= dataSize) ? J9UTF8_LENGTH(className) : LSW_STRING_MAX - 1;
+				copySize = (J9UTF8_LENGTH(className) < dataSize) ? J9UTF8_LENGTH(className) : (0 < dataSize ? (dataSize - 1) : 0);
 				memcpy(dataPtr, J9UTF8_DATA(className), copySize);
 				dataPtr[copySize] = 0;
```

Comment:
We may reach here with `dataSize == 0` which makes `dataPtr[copySize] = 0;` illegal. Suggest:
```c
				if (dataSize > 0) {
					className = J9ROMCLASS_CLASSNAME(class->romClass);
					copySize = (J9UTF8_LENGTH(className) < dataSize) ? J9UTF8_LENGTH(className) : (dataSize - 1);
					memcpy(dataPtr, J9UTF8_DATA(className), copySize);
					dataPtr[copySize] = 0;
				}
```

Reply:
Updated.